### ISSUE TITLE
poor result

### ISSUE THREAD

**Issue by jhy-dabeeo** [chenzhaiyu/points2poly#6]
Hi
I'm trying to make building models using your project.
but cannot get a decent result like your examples
can i get any advice or tip?

this is my input pointcloud building
![image](https://user-images.githubusercontent.com/69131502/193970997-a65154b1-c458-4c64-97e3-a23f0e3c23f5.png)

these are RANSAC parameters ( all default )
![image](https://user-images.githubusercontent.com/69131502/193971274-751f6963-6001-4e63-9d9f-1ca99fb1b0fe.png)

before detect Planars, i run Estimate normals with deafult setting 

this is the RANSAC result. it looks good to me
![image](https://user-images.githubusercontent.com/69131502/193971530-faaffd8b-af9c-45ff-8a0c-f5914ac2711d.png)

After saving the ply and vg files, run make_pc_dataset.py script and reconstruct.py
all parameters in config.yaml are default except normalize and append_bottom

cell_complex.visualize() result
![image](https://user-images.githubusercontent.com/69131502/193972385-345b19fd-5fc1-4b0f-a488-d3eda97cc263.png)

this is final result
All structures on the roof of the building have disappeared.
![image](https://user-images.githubusercontent.com/69131502/193972625-968398b6-4894-46d0-9acd-b642ca3555d7.png)

**Comment by chenzhaiyu** [comment]
Hi @jhy-dabeeo, three possible causes on top of my head:
1. The default [`coefficient`](https://github.com/chenzhaiyu/points2poly/blob/e5d2667a31cd1e42995037cc275f4476b6feafc2/conf/config.yaml#L25) value may not fit your data. Try to decrease the value to avoid _shrinking_ surfaces. The effect of this parameter is explained in section 5.2.4 of the [paper](https://arxiv.org/abs/2112.13142).
2.  Something possibly cracked with cell complex creation. From the `cell_complex.visualise()` screenshot I can't really see the cells from the slanted roof structure. Can you make sure they're there? You can also save the cells into PLM files with [`cell_complex.save_plm()`](https://github.com/chenzhaiyu/abspy/blob/b24823e5df6f3f4e004459657ad59af02c907a29/abspy/complex.py#L733), load them into Mapple, and switch on clipping-plane for better visualisation (Global - Clipping Plane).
3. Did you use the provided _Point2Surf_ model for SDF prediction? This pre-trained model may not fit your data (though it gets the overall building shape), in which case you need to retrain the model.

**Comment by jhy-dabeeo** [comment]
Thanks for replying
i tried several combinations of parameters.

1. RANSAC and coefficient
I changed minimum support of RANSAC Parameters from 1000 to 500 and dropped coeefficient value 0.0005
![image](https://user-images.githubusercontent.com/69131502/194446804-477a9894-5040-4b75-84e1-a285f6859f12.png)
![image](https://user-images.githubusercontent.com/69131502/194447387-613f1d01-ab4c-4fa7-b07c-4a389a53e4ca.png)

it looks way better then previous one but still not good to use as building model
![image](https://user-images.githubusercontent.com/69131502/194448647-8e3e01d6-3ce9-4b2a-abf9-2d24de23ed42.png)

2. cell_complex.save_plm()
![image](https://user-images.githubusercontent.com/69131502/194449466-28a701a3-f219-4073-8bb6-6f0ae13ce2c7.png)
![image](https://user-images.githubusercontent.com/69131502/194449486-4e26feb1-8b1a-4086-a79e-83b7602f5eb9.png)

3. pretrained model
i'im using model named helsinki_fullview which you provide

**Comment by chenzhaiyu** [comment]
Thanks for the follow-up. Now I can see the slanted roof structure in the cell complex. Your point cloud seems very noisy, which could result in 
1. excessive cells being generated from excessive plane segments,
2. the model pre-trained only with Gaussian noise being confused.

In this case, I'd guess possibly the combination of 1 and 2 resulted in the poor reconstruction result. Maybe try to tune the plane detection further (IMO this is actually the main limitation:  in the paper we assumed good-quality planes can be detected), or retrain a model with data of similar characteristics?

**Comment by chenzhaiyu** [comment]
Closed for now. Feel free to reopen in case of any updates.
